src/figuresClasses.ts

```diff
@@ -1,19 +1,125 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  color: Color;
 
-}
+  shape: Shape;
+
+  sideA: number;
+
+  sideB: number;
+
+  sideC: number;
 
-export class Circle {
+  constructor(
+    color: Color,
+    sideA: number,
+    sideB: number,
+    sideC: number,
+  ) {
+    this.shape = 'triangle';
```

Comment:
Let's set this value as class field, outside constructor function, no need to set it here